sdk/dotnet/Pulumi/Deployment/Deployment_Run.cs

```diff
@@ -56,6 +56,38 @@ public static Task<int> RunAsync(Func<IDictionary<string, object?>> func)
         /// in this dictionary will become the outputs for the Pulumi Stack that is created.
         /// </summary>
         public static Task<int> RunAsync(Func<Task<IDictionary<string, object?>>> func)
+            => CreateRunner().RunAsync(func);
+
+        /// <summary>
+        /// <see cref="RunAsync{TStack}()"/> is an
+        /// entry-point to a Pulumi application. .NET applications should perform all startup logic
+        /// they need in their <c>Main</c> method and then end with:
+        /// <para>
+        /// <c>
+        /// static Task&lt;int&gt; Main(string[] args)
+        /// {
+        ///     // program initialization code ...
+        ///
+        ///     return Deployment.Run&lt;MyStack&gt;();
+        /// }
+        /// </c>
+        /// </para>
+        /// <para>
+        /// Deployment will instantiate a new stack instance based on the type passed as TStack type parameter.
+        /// Importantly, cloud resources cannot be created outside of the <see cref="Stack"/> component.
+        /// </para>
+        /// <para>
+        /// Because cloud Resource construction is
+        /// inherently asynchronous, the result of this function is a <see cref="Task{T}"/> which should
+        /// then be returned or awaited.  This will ensure that any problems that are encountered during
+        /// the running of the program are properly reported.  Failure to do this may lead to the
+        /// program ending early before all resources are properly registered.
+        /// </para>
+        /// </summary>
+        public static Task<int> RunAsync<TStack>() where TStack : Stack, new()
+            => CreateRunner().RunAsync<TStack>();
+
+        private static IRunner CreateRunner()
```

Comment:
was breaking this out necessary?

Reply:
a convenience to save on copy-pasting. what's the downside?

Reply:
i'm still not getting it though. is this called from somewhere else?

Reply:
It's called from two overloads of `RunAsync`

Reply:
i'm so sorry.  i must have been blid.  i looked over and over and didn't see two calls.  now i do... i don't know waht went wrong with my brain.

Reply:
no worries!